File: food_ticket/frontend/src/types/order.ts
// src/types/order.ts
export interface OrderItem {
  product_id: number;
  quantity: number;
  unit_price: number;
}

export interface OrderCreate {
  store_id: number;
  attribute_id: number;  // 顔認証で取得したID（今は仮値）
  items: OrderItem[];
  total_amount: number;
  payment_method: string;
  take_out_type: string;
}

export interface OrderResponse {
  status: string;
  order_id: number;
}

File: food_ticket/backend/app/schemas/order.py
# app/schemas/order.py (または main.py 内)
from pydantic import BaseModel
from typing import List

class OrderItem(BaseModel):
    product_id: int
    quantity: int
    unit_price: int

class OrderCreate(BaseModel):
    store_id: int
    attribute_id: int  # 顔認証で取得したID
    items: List[OrderItem]
    total_amount: int
    payment_method: str
    take_out_type: str